src/App.tsx

```diff
@@ -15,47 +15,124 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      return !isReversed
+        ? visibleGoods.sort()
+        : visibleGoods.sort((a, b) => (a > b ? -1 : 1));
+    case SortType.LENGTH:
+      return !isReversed
+        ? visibleGoods.sort((a, b) => a.length - b.length)
+        : visibleGoods.sort((a, b) => a.length - b.length).reverse();
+    case SortType.NONE:
+      return isReversed ? visibleGoods.reverse() : visibleGoods;
+    default:
+      return visibleGoods;
+  }
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+  const [sortType, setSortType] = useState<SortType>(SortType.NONE);
+
+  const isInitialState = !isReversed && sortType === SortType.NONE;
+
+  const reverse = () => {
+    setIsReversed((state) => !state);
+  };
+
+  const alphSort = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const byLength = () => {
+    setSortType(SortType.LENGTH);
+  };
```

Comment:
pay attention to functions naming

<img width="1012" alt="Screenshot 2023-03-11 at 17 47 56" src="https://user-images.githubusercontent.com/32853034/224496929-07c0be63-5f70-4943-85e2-4f12736bb5e8.png">